Misc/NEWS.d/next/Library/2020-12-02-07-37-59.bpo-42532.ObNep_.rst

```diff
@@ -0,0 +1 @@
+Check if NonCallableMock.__new__'s spec_arg is not None instead of calling its __bool__ function
```

Comment:
Suggestion: I'd rephrase this to make it user-oriented. The user probably does not care if we compare to none, they would just see that we were unexpectedly calling the function.

```suggestion
Remove unexpected call of ``__bool__`` when passing a ``spec_arg`` argument to a Mock.
```